### PR TITLE
[feat] finalize onboarding styling

### LINKED ISSUE #48: Finalize Onboarding Styles + Create Onboarding Progress Bar
https://linear.app/trap-garden/issue/TG-40/finalize-onboarding-styles-create-progress-bar

remember to git checkout main → git pull → git checkout -b [thisbranch]

* assigned: Monday 11/18
* due (PR open by): Saturday 12/6
* MERGE by: Saturday 12/6

### PR DESCRIPTION
**Pull request by ccatherinetan**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
* Styled Onboarding Pages 
    * Created ProgressBar 
    * Consolidated styling in SelectionScreen
    * Moved Onboarding DropdownOptions to utils/helpers 
* Removed hasPlot/setHasPlot from ProfileContext 
* Refactored components
    * RadioGroup: ensure that no value is selected only if defaultValue is null (false should appear) 
    * CustomSelect: use placeholder and label props 
    * DateInput: add label prop 
    * Buttons: styling / fonts 
* Modified View Plants
    * Style Clear Filters Button 
    * Remove hasPlot from screen selection 
* Modified Add Details Flow
    * Used updated CustomSelect, DateInput in PlantDetails 
    * Refactored insertUserPlant query to insert userPlant array in 1 go. 

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

Onboarding: 
Screen Shot 2024-12-23 at 10 33 37 AM
Screen Shot 2024-12-23 at 10 33 42 AM
Onboarding3
Onboarding4

Add Details
AddDetails1
AddDetails_SelectOpen
AddDetails_Review

View Plants (see Clear Filters Button)
Screen Shot 2024-12-23 at 11 31 05 AM

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- garden set up guide 
- Planting timeline (key and month label) for plant pages 
- https://www.developerway.com/posts/react-re-renders-guide 
    - TODO: We don’t need to memoize the filtering? 
        - View Plants: filteredPlantList, filteredUserPlantList
        - Seasonal Planting Guide
- TODO: use CONFIG throughout the app 
- Consolidate Button styles
    - BigButton: P2 (14px) 
    - SmallButton: P3 (12px)
- Refactor CustomSelect to take in value:DropdownOption<T> rather than just T 
    - Need to change onboarding to take all DropdownOption<T> as states… 
    - Refactor DropdownOption<T> to type Record<T, string>, i.e. use the values as keys. 
- Separate out PlantContext 
    - we should prob separate setPlantsToAdd and plantsToAdd out into a separate context. otherwise, updating setPlantsToAdd would trigger a re-render of the entire app

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
 https://www.developerway.com/posts/react-re-renders-guide: on how to mitigate re-rendering

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ccatherinetan

### CODE REVIEW

File: app/onboarding/page.tsx [hunk 1]
```diff
@@ -56,74 +65,130 @@ interface ReviewPageProps {
...
-          </option>
-        ))}
-      </select>
-    </div>
+    <OnboardingContainer>
+      <Flex $direction="column" $align="center">
+        <ProgressBar progress={2.6} />
+        <P3 $color={COLORS.shrub} style={{ marginTop: '2.5rem' }}>
+          QUESTION 1 OF 3
+        </P3>
```
Comment (ccatherinetan) [inline app/onboarding/page.tsx]: since the progress bar and the question number are shared between multiple pages, we should move it outside of these individual components and into the OnboardingFlow component to reduce redundancy

If we end up making onboarding a multi-page flow next sem, these would be good candidates of things to include in the layout.tsx

So the individual question components can primarily just be the Question Title + the RadioOptions 

we might even consider creating a general component `OnboardingQuestion` that can be used for all 3 questions

File: app/onboarding/page.tsx [hunk 2]
```diff
@@ -56,74 +65,130 @@ interface ReviewPageProps {
...
+      </Flex>
+      <ButtonDiv>
+        <Button
+          onClick={onNext}
+          disabled={!selectedState}
+          $primaryColor={COLORS.shrub}
+        >
+          Next
+        </Button>
+      </ButtonDiv>
```
Comment (ccatherinetan) [inline app/onboarding/page.tsx]: i think it'd make sense to move the next/back buttons to OnboardingFlow as well

**Review by ccatherinetan — COMMENTED** [review]
looking amazing so far! I rebased + added some styling changes to OnboardingContainer, since I was assuming that we'd move OnboardingContainer out of the individual questions and into the OnboardingFlow component.

However, that resulted in the Review page being messed up. You can revert to your original style for OnboardingContainer if you'd like, but i think it's worth considering ways to simplify + reudce redundancy. 

Also, after rebasing, there were some issues with the style of the Progress bar against the Header, which we should address

File: app/onboarding/page.tsx [hunk 3]
```diff
@@ -162,34 +227,48 @@ const ReviewPage = ({
...
-          options={plotOptions}
-          onChange={value => setSelectedPlot(value === true)} // Convert the value as needed
-        />
-        <div style={{ height: '12.625rem' }} />
-        <BigButton color={COLORS.shrub} onClick={handleSubmit}>
-          Let&apos;s Start Growing !
-        </BigButton>
-      </ReviewContainer>
-    </PageContainer>
+          <LabeledCustomSelect
```
Comment (ccatherinetan) [inline app/onboarding/page.tsx]: note: instead of creating a new component, we want to just add a new prop to CustomSelect

File: app/onboarding/page.tsx [hunk 4]
```diff
@@ -162,34 +227,48 @@ const ReviewPage = ({
...
+            label="Plot Status"
+            value={selectedPlot}
+            options={plotOptions}
+            onChange={value => setSelectedPlot(value === true)} // Convert the value as needed
+          />
+        </ContentContainer>
+      </div>
+      <BigButton color={COLORS.shrub} onClick={handleSubmit}>
+        Let&apos;s Start Growing!
+      </BigButton>
```
Comment (ccatherinetan) [inline app/onboarding/page.tsx]: there should be both a "Back" and a "Let's Grow!" (Submit) button here. 
We can copy over the button styles from the other screens here.

File: app/onboarding/styles.ts [hunk 5]
```diff
@@ -1,45 +1,50 @@
 import styled from 'styled-components';
 import COLORS from '@/styles/colors';
 
-export const PageContainer = styled.div`
-  width: 100%;
-  min-height: 100vh;
-  background-color: ${COLORS.seed};
+export const OnboardingContainer = styled.div`
+  min-height: calc(100vh - 60px); // 60px is the hardcoded height of Header
```
Comment (ccatherinetan) [inline app/onboarding/styles.ts:5]: we should probably change this to use Flex `full-screen` so that there's a single source of truth in terms of the screen height

File: components/ProgressBar/styles.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import styled from 'styled-components';
+import COLORS from '@/styles/colors';
+
+export const BackgroundDiv = styled.div`
+  width: 100%;
+  height: 0.375rem;
```
Comment (ccatherinetan) [inline components/ProgressBar/styles.ts]: this should be px

File: components/CustomSelect/index.tsx [hunk 7]
```diff
@@ -48,31 +57,38 @@ const CustomSelect = <T extends string | number | boolean>({
...
+          {label}
+        </P2>
       )}
-    </SelectContainer>
+      <SelectContainer
+        ref={containerRef}
+        onClick={isContainerClickable ? () => setIsOpen(!isOpen) : undefined}
+      >
+        <P2 $color={COLORS.midgray}>
+          {options.find(option => option.value === value)?.label || placeholder}
```
Comment (ccatherinetan) [inline components/CustomSelect/index.tsx:74]: we may want to refactor `DropdownOption<T>` to `Record<T, string>`, i.e. use the values as keys, to mitigate needing to search through `options` for the corresponding label.

File: components/CustomSelect/index.tsx [hunk 8]
```diff
@@ -48,31 +57,38 @@ const CustomSelect = <T extends string | number | boolean>({
...
+        </P2>
+        <DropdownIcon id={componentId} onClick={() => setIsOpen(!isOpen)}>
+          <Icon type={iconType} />
+        </DropdownIcon>
+        {isOpen && (
+          <OptionsContainer>
+            {options.map(option => (
+              <Option
+                key={String(option.value)}
```
Comment (ccatherinetan) [inline components/CustomSelect/index.tsx:83]: the key can just be an index since the options list is static, but maybe if we foresee ever passing in dynamic options, this might be a good idea

File: utils/ProfileProvider.tsx [hunk 9]
```diff
@@ -19,11 +19,9 @@ import { useAuth } from './AuthProvider';
 export interface ProfileContextType {
   profileData: Profile | null;
   profileReady: boolean;
-  hasPlot: boolean | null;
   plantsToAdd: Plant[];
   setProfile: (completeProfile: Profile) => Promise<void>; // Now expects full Profile
   loadProfile: () => Promise<void>;
-  setHasPlot: (plotValue: boolean | null) => void;
   setPlantsToAdd: (plants: Plant[]) => void;
```
Comment (ccatherinetan) [inline utils/ProfileProvider.tsx:25]: we should prob separate `setPlantsToAdd` and `plantsToAdd` out into a separate context. otherwise, updating setPlantsToAdd would trigger a re-render of the entire app
